code/game/objects/items/robot/ai_upgrades.dm

```diff
@@ -1,4 +1,44 @@
 ///AI Upgrades
+/obj/item/aiupgrade
+	name = "ai upgrade disk"
+	desc = "You really shouldn't be seeing this"
+	icon = 'icons/obj/devices/circuitry_n_data.dmi'
+	icon_state = "datadisk3"
+	var/datum/ai_module/to_gift = /datum/ai_module
+
+/obj/item/aiupgrade/pre_attack(atom/A, mob/living/user, proximity)
+	if(!proximity)
+		return ..()
+	if(!isAI(A))
+		return ..()
+	var/mob/living/silicon/ai/AI = A
+	var/datum/action/innate/ai/action = locate(to_gift.power_type) in AI.actions
+	var/datum/ai_module/AM = new to_gift
```

Comment:
Name your variables something that is not just two capitalized letters (Except for the AI, which is already two latters. Common sense applies).